src/Operations.jl

```diff
@@ -1438,6 +1440,10 @@ function sandbox(fn::Function, ctx::Context, target::PackageSpec, target_path::S
             cp(sandbox_project, tmp_project)
             chmod(tmp_project, 0o600)
         end
+        if force_latest_compat
```

Comment:
Modify the `sandbox_project_override.env.project.compat` before it is written (at line 1437) and the correct compat should just be printed automatically. No need to write to disk, chmod, read it back in, modify compat, write it back out to disk, chmod again.

Reply:
That works if `sandbox_project_override !== nothing`. But what if we are in the other branch, i.e. `isfile(sandbox_project)`?